packages/app/src/app/overmind/namespaces/sidebar/actions.ts

```diff
@@ -0,0 +1,50 @@
+import { Context } from 'app/overmind';
+
+export const getSidebarData = async (
+  { state, effects }: Context,
+  teamId?: string
+) => {
+  const {
+    gql: { queries },
+  } = effects;
+
+  try {
+    if (teamId) {
+      /**
+       * Fetch data for the selected team
+       */
+      const {
+        me: {
+          team: { sandboxes, templates },
+        },
+      } = await queries.getTeamSidebarData({ id: teamId });
+
+      const hasSyncedSandboxes = sandboxes?.length > 0;
+      const hasTemplates = templates?.length > 0;
+
+      state.sidebar = {
+        hasSyncedSandboxes,
+        hasTemplates,
+      };
+    } else {
+      /**
+       * Fetch data for the user
+       */
+      const {
+        me: { sandboxes, templates },
+      } = await queries.getPersonalSidebarData();
```

Comment:
nitpick: this if/else looks like it can be made around the call of `queries.getPersonalSidebarData` vs `queries.getTeamSidebarData`, but the rest of the function can be common. would keep the function shorter and less repetitive

Reply:
I remember doing it this way because the types were acting up, but since they still were I tried to implement it in another way. I've changed it but its not a lot shorter because I had to do some workarounds for TS. It was weird to me that my editor didn't show errors for certain things, but running `yarn typecheck` does. Something fishy.